### PR TITLE
Various stat collection tweaks

### PR DESCRIPTION
**Pull request by gbasood**
~~Fixes a problem where a user's key would have a linebreak and break stat parsing.~~ That problem isn't even related to stats wooooo
Adds stats for blobs including victory because for SOME REASON I didn't think that part through.

### CODE REVIEW

**Comment by gbasood** [comment]
p.s. stat serb is growing

File: code/datums/statistics/stat_blob.dm [hunk 1]
```diff
@@ -31,3 +31,12 @@
...
 /datum/stat_blob/xeno/writeStats(file)
 	file << "XENOSTATS|[eggs_laid]|[faces_hugged]|[proper_head_protection]"
+
+/datum/stat_blob/blobmode
+	var/blob_wins = 0
+	var/spawned_blob_players = 0
+	var/spores_spawned = 0
+	var/res_generated = 0 // Note this does not take into consideration points that were wasted for going over the maximum possible points
+
+/datum/stat_blob/blobmode/writeStats()
```
Comment (ComicIronic) [inline code/datums/statistics/stat_blob.dm]: Missing a parameter there. How is this compiling?

Comment (gbasood) [inline code/datums/statistics/stat_blob.dm]: It's not compiling that's how

File: code/game/gamemodes/blob/blob_finish.dm [hunk 2]
```diff
@@ -2,6 +2,7 @@
 	if(!declared)//No blobs have been spawned yet
 		return 0
 	if(blobwincount <= blobs.len)//Blob took over
+		stat_collection.blob_wins = 1
```
Comment (ComicIronic) [inline code/game/gamemodes/blob/blob_finish.dm]: blobblob.blob_wins

**Pull request merged**